spec/rubocop/cop/performance/chain_array_allocation_spec.rb

```diff
@@ -53,4 +53,21 @@
       RUBY
     end
   end
+
+  describe 'when using `select` with block argument after `select`' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        model.select(:foo, :bar).select { |item| item.do_something }
+                                ^^^^^^^ Use unchained `select` and `select!` (followed by `return array` if required) instead of chaining `select...select`.
+      RUBY
+    end
+  end
```

Comment:
Forgive my possible stupidity (I'm not terribly familiar with how the tests in rubocop work)
But wouldn't this still be wrong?
Wouldn't it flag `select` (QueryMethods#select) followed by a `select` (enumerable).
Which isn't a double select that requires optimization?

You would however want to flag the 3rd select here right?
`model.select(:foo, :bar).select { |item| item.do_something }.select { |item| item.do_something_else }`

Reply:
I'm not entirely sure, this patch works as follows:

```ruby
it 'registers an offense' do
  expect_offense(<<~RUBY)
    model.select(:foo, :bar).select { |item| item.do_something }.select { |item| item.do_something_else }
                            ^^^^^^^ Use unchained `select` and `select!` (followed by `return array` if required) instead of chaining `select...select`.
                                                                ^^^^^^^ Use unchained `select` and `select!` (followed by `return array` if required) instead of chaining `select...select`.
  RUBY
end
```

So, this is a patch that distinguishes between `select(:arg)` and `select { |x| x }`. What potential issues might arise from this?

Reply:
So, let me rephrase
with this code 
```
expect_offense(<<~RUBY)
        model.select(:foo, :bar).select { |item| item.do_something }
                                ^^^^^^^ Use unchained `select` and `select!` (followed by `return array` if required) instead of chaining `select...select`.
RUBY
```
What should the suggested corrected code look like?

Reply:
Ah, I understand. This should definitely be recognized as a false positive and I've fixed it. Thank you!